common/src/main/java/com/microsoft/identity/common/internal/controllers/BrokerMsalController.java

```diff
@@ -484,6 +593,75 @@ public void putValueInSuccessEvent(final @NonNull ApiEndEvent event, final @NonN
                 });
     }
 
+    /**
+     * Get the intent for the broker ATv2 request
+     *
+     * @param parameters a {@link ATv2TokenCommandParameters}
+     * @return an {@link Intent} for initiating Broker ATv2 interactive activity.
+     */
+    private @NonNull
+    Intent getBrokerAuthorizationIntentForATv2(
+            final @NonNull ATv2TokenCommandParameters parameters) throws BaseException {
+        return getBrokerOperationExecutor().execute(parameters,
+                new BrokerOperation<Intent>() {
+                    private String negotiatedBrokerProtocolVersion;
+
+                    @Override
+                    public void performPrerequisites(final @NonNull IIpcStrategy strategy) throws BaseException {
+                        verifyTokenParametersAreSupported(parameters);
+                        negotiatedBrokerProtocolVersion = hello(strategy, parameters.getRequiredBrokerProtocolVersion());
+                    }
+
+                    @Override
+                    public @NonNull
+                    BrokerOperationBundle getBundle() {
+                        return new BrokerOperationBundle(
+                                MSAL_GET_INTENT_FOR_ATV2_INTERACTIVE_REQUEST,
+                                mActiveBrokerPackageName,
+                                null);
+                    }
+
+                    // TODO: Confused, why is this called "extractResultBundle", but we're actually
+                    //  returning an intent where we create the bundle for the request?
+                    //  Is this the result of performPrerequisites with broker?
```

Comment:
Ping

Reply:
This for extract the result obtained from broker (resultBundle) into object you want. In this case it's Intent object (BrokerOperation<Intent>()  {} above)